src/luarocks/build.lua

```diff
@@ -303,26 +303,26 @@ function build.build_rockspec(rockspec_file, need_to_fetch, minimal_mode, deps_m
       if fs.is_dir(copy_dir) then
          local dest = dir.path(path.install_dir(name, version), copy_dir)
          fs.make_dir(dest)
-         fs.copy_contents(copy_dir, dest)
+         fs.copy_contents(copy_dir, dest, "read")
```

Comment:
I am concerned about the behavior of `copy_contents`: since a tree of directories contains a mix of readable and executable files (plus all subdirectories themselves need to be executable), the `copy_contents` function needs a mode where it preserves each file's permissions (in other words, the third argument should really be optional). 

I think you'll need to make revert most (all?) places where `copy_contents` was being called with 2 args only.